openrouteservice/src/main/java/heigit/ors/config/AppConfig.java

```diff
@@ -51,6 +55,21 @@ public AppConfig()	{
     	
     	File file = new File(url.getPath());
 		_config = ConfigFactory.parseFile(file);
+
+		//Modification by H Leuschner: Save md5 hash of map file in static String for access with every request
+		File graphsDir = new File("graphs");
+		File[] md5Files = graphsDir.listFiles(new FileFilter() {
+			public boolean accept(File pathname) {
+				return pathname.getName().endsWith(".md5");
+			}
+		});
+		if (md5Files.length == 1){
+			try{
+				osm_md5_hash = FileUtility.readFile(md5Files[0].toString()).trim();
+			}
+			catch (IOException e)
+			{}
```

Comment:
It might be worth adding an error message to the log if there is a problem reading the md5 files seeing as this is a new feature and so would be easier for debugging. It doesn't need to be sent to the user, just logged.